wallets/provider-metamask/src/helpers.ts

```diff
@@ -1,17 +1,81 @@
-import { getCoinbaseInstance } from '@rango-dev/wallets-shared';
+function getMetaMask(ethereum: any) {
+  const isMetaMask = !!ethereum?.isMetaMask;
+  if (!isMetaMask) {
+    return;
+  }
+  /*
+   * Brave tries to make itself look like MetaMask
+   * Could also try RPC `web3_clientVersion` if following is unreliable
+   */
+  if (ethereum.isBraveWallet && !ethereum._events && !ethereum._state) {
+    return;
+  }
+  if (ethereum.isApexWallet) {
+    return;
+  }
+  if (ethereum.isAvalanche) {
+    return;
+  }
+  if (ethereum.isBitKeep) {
+    return;
+  }
+  if (ethereum.isBlockWallet) {
+    return;
+  }
+  if (ethereum.isCoin98) {
+    return;
+  }
+  if (ethereum.isFordefi) {
+    return;
+  }
+  if (ethereum.__XDEFI) {
+    return;
+  }
+  if (ethereum.isMathWallet) {
+    return;
+  }
+  if (ethereum.isOkxWallet || ethereum.isOKExWallet) {
+    return;
+  }
+  if (ethereum.isOneInchIOSWallet || ethereum.isOneInchAndroidWallet) {
+    return;
+  }
+  if (ethereum.isOpera) {
+    return;
+  }
+  if (ethereum.isPortal) {
+    return;
+  }
+  if (ethereum.isRabby) {
+    return;
+  }
+  if (ethereum.isDefiant) {
+    return;
+  }
+  if (ethereum.isTokenPocket) {
+    return;
+  }
+  if (ethereum.isTokenary) {
+    return;
+  }
+  if (ethereum.isZeal) {
+    return;
+  }
+  if (ethereum.isZerion) {
+    return;
+  }
+  return ethereum;
+}
 
 export function metamask() {
-  const isCoinbaseWalletAvailable = !!getCoinbaseInstance();
   const { ethereum } = window;
 
-  // Some wallets overriding the metamask. So we need to get it properly.
-  if (isCoinbaseWalletAvailable) {
-    // Getting intance from overrided structure from coinbase.
-    return getCoinbaseInstance('metamask');
-  } else {
-    if (!!ethereum && ethereum.isMetaMask) {
-      return ethereum;
+  if (Array.isArray(ethereum?.providers)) {
+    let providers = ethereum.providers;
+    if (Array.isArray(ethereum.providers[0]?.providers)) {
+      providers = ethereum.providers[0]?.providers;
     }
+    return providers.find(getMetaMask) ?? null;
   }
-  return null;
+  return ethereum.isMetaMask ? ethereum : null;
```

Comment:
This code throws error when there is no metamask extension enabled.
```
TypeError: Cannot read properties of undefined (reading 'isMetaMask')
```

```suggestion
  return ethereum?.isMetaMask ? ethereum : null;
```